quantmsutils/diann/diann2msstats.py

```
@@ -116,21 +116,58 @@ def get_exp_design_dfs(exp_design_file):
+    return s_data_frame, f_table
+
+
+def _parse_legacy_design(data, exp_design_file):
+    """Parse the legacy two-table experimental design format (blank line separator)."""
+    try:
+        empty_row = data.index("\n")
+    except ValueError:
+        raise ValueError(
+            f"Could not find blank separator row in {exp_design_file}. "
+            "Ensure the file contains a blank line between the file and sample tables."
+        )
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)